crates/csharp/src/function.rs

```diff
@@ -754,7 +754,7 @@ impl Bindgen for FunctionBindgen<'_, '_> {
                                 "
                             );
                         }
-                        results.push(format!("(nint){ptr}"));
+                        results.push(format!("(int){ptr}"));
```

Comment:
why this change?

Reply:
Because it generates 
```cs
BitConverter.TryWriteBytes(new Span<byte>((void*)(basePtr + 8), 4), (nint)listPtr);
```

And that uses `TryWriteBytes(Span<Byte>, UInt64) ` which is wrong. There is no TryWriteBytes overload for `nint`.
And then it returns `false` but we ignore it.
Let's stop using TryWriteBytes. 

Instead it could be just 
```cs
new Span<nint>((void*)(basePtr + 8), 1)[0] = listPtr;
```